ceph_iscsi_config/lun.py

```diff
@@ -605,6 +606,7 @@ def allocate(self, keep_dev_in_lio=True, in_wwn=None):
 
                 if not rbd_image.error:
                     self.config.add_item('disks', self.config_key)
+                    need_del_item = True
```

Comment:
What is the point of adding a partially-formed config item before the operation actually succeeds?  Instead of attempting to clean up on all error paths except one, would delaying the addition until it is known that the config item is actually needed work?